Dockerfile

```diff
@@ -2,22 +2,18 @@
 FROM microsoft/aspnetcore-build:2.0 AS build-env
 WORKDIR /build
 
-# Restore dependencies
-COPY src/ServiceHost/ServiceHost.csproj ./src/ServiceHost/
-COPY test/ServiceHost.Tests/ServiceHost.Tests.csproj ./test/ServiceHost.Tests/
-
-RUN dotnet restore src/ServiceHost/ServiceHost.csproj
-RUN dotnet restore test/ServiceHost.Tests/ServiceHost.Tests.csproj
-
 # Copy source
 COPY . .
 
+# Restore dependencies
+RUN dotnet restore src/ServiceHost/ServiceHost.csproj /p:SolutionDir=/build /p:SolutionName=DotNetRu.Server
+RUN dotnet restore test/ServiceHost.Tests/ServiceHost.Tests.csproj /p:SolutionDir=/build /p:SolutionName=DotNetRu.Server
+
 # Run tests
-RUN dotnet test test/ServiceHost.Tests/ServiceHost.Tests.csproj
+RUN dotnet test test/ServiceHost.Tests/ServiceHost.Tests.csproj /p:SolutionDir=/build /p:SolutionName=DotNetRu.Server
 
 # Publish stage
-RUN dotnet publish src/ServiceHost/ServiceHost.csproj -o /publish
-
+RUN dotnet publish src/ServiceHost/ServiceHost.csproj -o /publish /p:SolutionDir=/build /p:SolutionName=DotNetRu.Server
 
 ## Runtime stage
 FROM microsoft/aspnetcore:2.0
```

Comment:
Change to latest version Change image to latest version microsoft/dotnet:2.1-aspnetcore-runtime